packages/react-components/react-dialog/src/components/Dialog/Dialog.types.ts

```diff
@@ -1,18 +1,47 @@
 import type * as React from 'react';
 import type { ComponentProps, ComponentState } from '@fluentui/react-utilities';
 import type { DialogContextValue, DialogSurfaceContextValue } from '../../contexts';
+import type { DialogSurfaceElement } from '../DialogSurface/DialogSurface.types';
 
 export type DialogSlots = {};
 
-export type DialogOpenChangeEvent = React.KeyboardEvent | React.MouseEvent;
+export type DialogOpenChangeEvent = DialogOpenChangeData['event'];
 
 export type DialogOpenChangeData =
-  | { type: 'escapeKeyDown'; open: boolean; event: React.KeyboardEvent }
-  | { type: 'backdropClick'; open: boolean; event: React.MouseEvent }
-  | { type: 'triggerClick'; open: boolean; event: React.MouseEvent };
+  | {
+      /**
+       * triggered when Escape key is pressed in a native `dialog`
+       */
+      type: 'dialogCancel';
+      open: boolean;
+      event: React.SyntheticEvent<DialogSurfaceElement>;
+    }
```

Comment:
Because on native `dialog` pressing escape emits a [cancel event](https://developer.mozilla.org/en-US/docs/Web/API/HTMLDialogElement/cancel_event), which would be a simple `Event` instead of a `KeyboardEvent`, making both types incompatible.